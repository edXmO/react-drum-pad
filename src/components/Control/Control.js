import React, { useState } from 'react';
import '../../css/App.css';
import Display from './Display/Display';
import Panel from './Panel/Panel';
import Copy from './Copy/Copy';

const Control = () => {

    const [toggle, setToggle] = useState(false);
    const [volume, setVolume] = useState(80);

    const handleToggle = () => {
        setToggle(prevState => !prevState);
    }


    return (
        <div className='control'>
            <Display type={`VOL: ${volume}%`} screenType={`display display__volume ${toggle ? 'display display__volume--on' : ''}`} />
            <Panel toggle={toggle} setVolume={setVolume} handleToggle={handleToggle} />
            <Display type={'Hi hat'} screenType={`display display__sound ${toggle ? 'display display__sound--on' : ''}`} />
            <Copy />
        </div>
    )
}

export default Control;