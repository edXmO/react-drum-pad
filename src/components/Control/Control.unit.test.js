import React from 'react';
import renderer from 'react-test-renderer';

// Component
import Control from './Control';

describe('should render without crashing', () => {
    let component = renderer.create(
        <Control />
    );
    it('should render without crashing', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
});