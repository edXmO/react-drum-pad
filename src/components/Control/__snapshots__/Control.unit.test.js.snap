// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing should render without crashing 1`] = `
<div
  className="control"
>
  <div
    className="display display__volume "
  >
    VOL: undefined%
  </div>
  <div
    className="btn-panel"
  >
    <div
      className="btn-panel__box "
    >
      <button
        className="btn-panel__btn"
        onClick={[Function]}
      />
      OFF
    </div>
    <div
      className="btn-panel__box "
    >
      <button
        className="btn-panel__btn"
        onClick={[Function]}
      />
      +VOL
    </div>
    <div
      className="btn-panel__box "
    >
      <button
        className="btn-panel__btn"
        onClick={[Function]}
      />
      -VOL
    </div>
  </div>
  <div
    className="display display__sound "
  />
  <div
    className="copy"
  >
    Built by edXmO Â© 2021
  </div>
</div>
`;
